e2e/testcases/reconciler_finalizer_test.go

```diff
@@ -522,8 +522,18 @@ func TestReconcileFinalizerReconcileTimeout(t *testing.T) {
 	if err := nt.WatchForAllSyncs(); err != nil {
 		nt.T.Fatal(err)
 	}
+	nt.T.Cleanup(func() {
+		// Remove the fake finalizer. This should be already handled in the happy
+		// path, but remove in case there was an error.
+		namespace := fake.NamespaceObject(namespaceNN.Name)
```

Comment:
If the test succeeded, this would re-create the namespace. I don't think Apply returns NotFound. You could use Update tho. I don't think it needs SSA particularly.

Reply:
Ah good point, I'll fix this

Reply:
Done